pkg/volume/csi/csi_client.go

```diff
@@ -177,13 +179,25 @@ func (c *csiDriverClient) NodeGetInfo(ctx context.Context) (
 	accessibleTopology map[string]string,
 	err error) {
 	klog.V(4).Info(log("calling NodeGetInfo rpc"))
-	if c.nodeV1ClientCreator != nil {
-		return c.nodeGetInfoV1(ctx)
-	} else if c.nodeV0ClientCreator != nil {
-		return c.nodeGetInfoV0(ctx)
-	}
 
-	err = fmt.Errorf("failed to call NodeGetInfo. Both nodeV1ClientCreator and nodeV0ClientCreator are nil")
+	// TODO retries should happen at a lower layer (issue #73371)
+	backoff := wait.Backoff{Factor: 2.0, Steps: 5}
+	err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+		var getNodeInfoError error
+		if c.nodeV1ClientCreator != nil {
+			nodeID, maxVolumePerNode, accessibleTopology, getNodeInfoError = c.nodeGetInfoV1(ctx)
+		} else if c.nodeV0ClientCreator != nil {
+			nodeID, maxVolumePerNode, accessibleTopology, getNodeInfoError = c.nodeGetInfoV0(ctx)
+		}
+		if nodeID != "" {
```

Comment:
Maybe check for error explicitly instead (`getNodeInfoError == nil)? Because `nodeId` may accidentally be set even on error?

```GO
if getNodeInfoError == nil {
  return true, nil
}
```

Reply:
The reason I didn't check for error is because I thought we wanted to retry if we got back an error (unless it's a terminal error)?

Reply:
Yes we do want to retry if we get an error except on terminal errors. But we should check the error instead of the result to handle the case mentioned above. That said, not a huge deal since we can see that `nodeGetInfo...` doesn't return non-empty nodeID with non-nil err (at least for now), and this code will be removed in the future.